hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ListIterator.java

```diff
@@ -0,0 +1,340 @@
+
+    public int compareTo(HeapEntry other) {
+      return Comparator.comparing(HeapEntry::getKey)
+          .thenComparing(HeapEntry::getEntryIteratorId).compare(this, other);
+    }
+
+    public boolean equals(Object other) {
+      if (other == null) {
+        return false;
+      }
+
+      if (!(other instanceof HeapEntry)) {
+        return false;
+      }
+
+
+      HeapEntry that = (HeapEntry) other;
+      return this.compareTo(that) == 0;
+    }
+
+    public int hashCode() {
+      return key.hashCode();
+    }
+  }
+
+  /**
+   * Iterator for DB entries from a given rocksDB table.
+   */
+  public static class DbTableIter<Value> implements
+      ClosableIterator {
+    private final int entryIteratorId;
+    private final TableIterator<String,
+        ? extends Table.KeyValue<String, Value>> tableIterator;
+
+    private final Table<String, Value> table;
+    private HeapEntry currentKey;
+
+    DbTableIter(int entryIteratorId, Table<String, Value> table,
+                String prefixKey, String startKey) throws IOException {
+      this.entryIteratorId = entryIteratorId;
+      this.table = table;
+      this.tableIterator = table.iterator(prefixKey);
+      this.currentKey = null;
+
+      // only seek for the start key if the start key is lexicographically
+      // after the prefix key. For example
+      // Prefix key = 1024/c, Start key = 1024/a
+      // then do not seek for the start key
+      //
+      // on the other hand,
+      // Prefix key = 1024/a, Start key = 1024/c
+      // then seek for the start key
+      if (!StringUtils.isBlank(startKey) &&
+          startKey.compareTo(prefixKey) > 0) {
+        tableIterator.seek(startKey);
+      }
+    }
+
+    private void getNextKey() throws IOException {
+      while (tableIterator.hasNext() && currentKey == null) {
+        Table.KeyValue<String, Value> entry = tableIterator.next();
+        String entryKey = entry.getKey();
+        if (!KeyManagerImpl.isKeyInCache(entryKey, table)) {
+          currentKey = new HeapEntry(entryIteratorId,
+              table.getName(), entryKey, entry.getValue());
+        }
+      }
+    }
+
+    public boolean hasNext() {
+      try {
+        getNextKey();
+      } catch (Throwable t) {
+        throw new NoSuchElementException();
+      }
+      return currentKey != null;
+    }
+
+    public HeapEntry next() {
+      if (hasNext()) {
+        HeapEntry ret = currentKey;
+        currentKey = null;
+        return ret;
+      }
+      throw new NoSuchElementException();
+    }
+
+    public void close() throws IOException {
+      tableIterator.close();
+    }
+  }
+
+  /**
+   * Iterator for Cache entries in a Dir and File Table.
+   */
+  public static class CacheIter<Value>
+      implements ClosableIterator {
+    private final Map<String, Value> cacheKeyMap;
+
+    private final Iterator<Map.Entry<String, Value>>
+        cacheCreatedKeyIter;
+    private final Iterator<Map.Entry<CacheKey<String>, CacheValue<Value>>>
```

Comment:
nit: There is no needed to keep `cacheIter`. You can pass it to `getCacheValues()` and initialize `cacheKeyMap`. `populateCacheMap` would be better name than `getCacheValues`

Reply:
done